api/v1beta1/types.go

```diff
@@ -862,6 +862,10 @@ type APIServerLoadBalancer struct {
 	// AvailabilityZone is the failure domain that will be used to create the APIServerLoadBalancer Spec.
 	//+optional
 	AvailabilityZone optional.String `json:"availabilityZone,omitempty"`
+
+	// Flavor is the flavor name that will be used to create the APIServerLoadBalancer Spec.
+	//+optional
+	Flavor optional.String `json:"flavor,omitempty"`
```

Comment:
I'm fine with this change in the current state of things. However I know Matt has been thinking/designing API endpoints with different providers; so we might want to not add more fields here. I have no strong opinion, just tagging @mdbooth for now.